packages/theming/src/utils/withTheme.js

```diff
@@ -6,8 +6,15 @@
  */
 
 import { withTheme as styledWithTheme } from 'styled-components';
+import { default as theme } from '../theme';
 
 /** @component */
 export default function withTheme(WrappedComponent) {
+  if (WrappedComponent.defaultProps === undefined) {
```

Comment:
Would this be best enforced at the component level? There might be use-cases where we provide a tweaked default theme for a specific component.

Reply:
This just ensures that `withTheme` provides the same information to a Component as we've agreed will be provided to a StyledComponent (either through ThemeProvider context or via defaultProps). The code here would still allow providing

> a tweaked default theme for a specific component

I've added a test showing that `withTheme(<Component theme={theme} />)` works as expected.